sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/sources/ForeachWriterSuite.scala

```diff
@@ -303,6 +347,13 @@ class TestForeachWriter extends ForeachWriter[Int] {
 
   override def close(errorOrNull: Throwable): Unit = {
     events += ForeachWriterSuite.Close(error = Option(errorOrNull))
+    if (errorOrNull == null) {
```

Comment:
Why only if no error?

Reply:
since now abort is invoked after the close, i would like  make sure that the events are not added twice